tests/unit/amo/sagas/test_addonsByAuthors.js

```diff
@@ -0,0 +1,108 @@
+import SagaTester from 'redux-saga-tester';
+
+import addonsByAuthorsReducer, {
+  fetchOtherAddonsByAuthors,
+  loadOtherAddonsByAuthors,
+} from 'amo/reducers/addonsByAuthors';
+import addonsByAuthorsSaga from 'amo/sagas/addonsByAuthors';
+import { ADDON_TYPE_THEME, CLEAR_ERROR } from 'core/constants';
+import * as searchApi from 'core/api/search';
+import apiReducer from 'core/reducers/api';
+import {
+  createAddonsApiResult,
+  dispatchClientMetadata,
+  fakeAddon,
+} from 'tests/unit/amo/helpers';
+import { createStubErrorHandler } from 'tests/unit/helpers';
+
+
+describe(__filename, () => {
+  let errorHandler;
+  let mockApi;
+  let sagaTester;
+
+  beforeEach(() => {
+    errorHandler = createStubErrorHandler();
+    mockApi = sinon.mock(searchApi);
+    sagaTester = new SagaTester({
+      initialState: dispatchClientMetadata().state,
+      reducers: {
+        api: apiReducer,
+        addonsByAuthors: addonsByAuthorsReducer,
+      },
+    });
+    sagaTester.start(addonsByAuthorsSaga);
+  });
+
+  function _fetchOtherAddonsByAuthors(params) {
+    sagaTester.dispatch(fetchOtherAddonsByAuthors({
+      errorHandlerId: errorHandler.id,
+      addonType: ADDON_TYPE_THEME,
+      ...params,
+    }));
+  }
+
+  it('calls the API to retrieve other add-ons', async () => {
+    const addons = [fakeAddon];
+    const slug = fakeAddon.slug;
+    const authors = ['mozilla', 'johnedoe'];
+
+    mockApi
+      .expects('search')
```

Comment:
This should add some expectations about the arguments given to `search()`. This will add better coverage for the action creator payload and how the saga handles it.